File: TypeScript/src/accounts/paymentaccount.ts
abstract class PaymentAccount {
	private accountName: string;
	private accountNumber: number;

	constructor(accountName: string, accountNumber: number) {
		this.accountName = accountName;
		this.accountNumber = accountNumber;
	}

	public abstract pay(amount: number): boolean;

	public getAccountName(): string {
		return this.accountName;
	}

	public getAccountNumber(): number {
		return this.accountNumber;
	}
}

export { PaymentAccount }

File: Java/src/main/java/account/PaymentAccount.java
package account;

public abstract class PaymentAccount {

    private final String accountHolder;
    private final int accountNumber;

    public PaymentAccount(String accountHolder, int accountNumber) {
        this.accountHolder = accountHolder;
        this.accountNumber = accountNumber;
    }

    public abstract boolean pay(int amount);

    public int getAccountNumber() {
        return this.accountNumber;
    }

    public String getAccountHolder() {
        return this.accountHolder;
    }
}
